### PR TITLE
Use creep for heal

### PR DESCRIPTION
**Pull request by bmorel**

### CODE REVIEW

**Comment by bmorel** [comment]
Rebased to share code with #1698

**Comment by bmorel** [comment]
I added a (separated) commit to disable the work around lack of resupply API, since I have also a resupply API and usage in PRs

File: src/sgame/sg_bot_ai.cpp [hunk 1]
```diff
@@ -1109,10 +1109,10 @@ static AINodeStatus_t BotActionReachHealA( gentity_t *self )
 	int targetType = self->botMind->goal.getTargetedEntity()->s.modelindex;
 	vec3_t mins, maxs;
 	BG_BuildableBoundingBox( targetType, mins, maxs );
-	float dist = fabs( mins[0] );
+	float dist = RadiusFromBounds2D( mins, maxs );
```
Comment (slipher) [inline src/sgame/sg_bot_ai.cpp:1107]: Creep and booster healing are based on center-to-center distance between the entities, as is GoalInRange. So bounding box sizes are not relevant and shouldn't be considered.

**Comment by bmorel** [comment]
Building radius is no longer used

File: src/sgame/sg_bot_nav.cpp [hunk 2]
```diff
@@ -104,7 +104,7 @@ Bot Navigation Querys
 ========================
 */
 
-static float RadiusFromBounds2D( const vec3_t mins, const vec3_t maxs )
+float RadiusFromBounds2D( const vec3_t mins, const vec3_t maxs )
```
Comment (slipher) [inline src/sgame/sg_bot_nav.cpp:107]: This commit should be dropped now

Comment (bmorel) [reply src/sgame/sg_bot_nav.cpp]: done.

File: src/sgame/sg_bot_ai.cpp [hunk 3]
```diff
@@ -1100,7 +1100,17 @@ static AINodeStatus_t BotActionReachHealA( gentity_t *self )
 		return STATUS_FAILURE;
 	}
 
-	if ( !GoalInRange( self, 100 ) )
+	// retrieve creep size or booster's range to have proper distance
+	int targetType = self->botMind->goal.getTargetedEntity()->s.modelindex;
+	vec3_t mins, maxs;
+	BG_BuildableBoundingBox( targetType, mins, maxs );
```
Comment (slipher) [inline src/sgame/sg_bot_ai.cpp:1106]: There is still useless bounding box stuff

File: src/sgame/sg_bot_ai.cpp [hunk 4]
```diff
@@ -1100,7 +1100,17 @@ static AINodeStatus_t BotActionReachHealA( gentity_t *self )
...
+	float dist = BG_Buildable( targetType )->creepSize - 1.f; // minus one to make sure to step on creep
+	if ( targetType == BA_A_BOOSTER )
+	{
+		dist += REGEN_BOOSTER_RANGE;
```
Comment (slipher) [inline src/sgame/sg_bot_ai.cpp:1110]: The distances should not be added together. Rather take the maximum

**Comment by bmorel** [comment]
Comments have been addressed.

**Comment by slipher** [comment]
Well that's strangely written with `+=` instead of `=`, but it works I guess. LGTM

**Comment by bmorel** [comment]
true. I fixed it, will merge after the tests pass.

File: src/sgame/sg_bot_ai.cpp [hunk 5]
```diff
@@ -1100,7 +1100,19 @@ static AINodeStatus_t BotActionReachHealA( gentity_t *self )
...
+	// retrieve creep size or booster's range to have proper distance
+	int targetType = self->botMind->goal.getTargetedEntity()->s.modelindex;
+	float dist = 0;
+	if ( targetType != BA_A_BOOSTER )
+	{
+		dist = BG_Buildable( targetType )->creepSize - 1.f; // minus one to make sure to step on creep
+	}
+	else
+	{
+		dist = REGEN_BOOSTER_RANGE;
```
Comment (necessarily-equal) [inline src/sgame/sg_bot_ai.cpp:1112]: I kinda wonder why you don't add a `-1.0f` here, but LGTM otherwise

Comment (bmorel) [reply src/sgame/sg_bot_ai.cpp:1112]: good catch

**Review by necessarily-equal — APPROVED** [review]

**Pull request merged**